src/memote_webservice/resources/report.py

```diff
@@ -63,18 +62,17 @@ class Report(Resource):
 
     def get(self, uuid):
         """Return a snapshot report as JSON or HTML based on Accept headers."""
-        LOGGER.debug("Create connection to '%s'.", app.config["REDIS_URL"])
-        with Connection(redis.from_url(app.config["REDIS_URL"])):
-            LOGGER.debug("Using queue '%s'.", app.config["QUEUES"][0])
-            queue = Queue(app.config["QUEUES"][0])
-            job = queue.fetch_job(uuid)
-        if job is None:
-            msg = f"Result {uuid} does not exist."
-            LOGGER.info(msg)
-            api.abort(404, msg)
-        if job.is_finished:
+        result = AsyncResult(id=uuid, app=celery_app)
+        if result.ready():
             # Extract the SnapshotReport object's result attribute.
-            report = job.result
+            try:
```

Comment:
What is your take on handling any undefined exception? And how do you feel about this entire module not being unit tested? Codecov specifically complains about lines 63-80. I'm not sure how necessary it is but I think I'd feel safer knowing this is covered rigidly.

Reply:
This is fine then!